### PR TITLE
Modid

### PR DESCRIPTION
**Pull request by thattil**
-serialnumber detached from eiger
-moduleid get for eiger, m3 and g2

they all use detid_[detectortype] files. (eiger uses detid.txt for real hardware as it was in production).
get will get from file. ( set into register at server startup)

### CODE REVIEW

File: slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c [hunk 1]
```diff
@@ -322,15 +302,22 @@ u_int32_t getDetectorIP() {
 void initControlServer() {
     LOG(logINFOBLUE, ("Configuring Control server\n"));
     if (!updateFlag && initError == OK) {
-        readDetectorNumber();
+#ifndef VIRTUAL
```
Comment (erikfrojdh) [inline slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c:305]: Why is there a difference between virtual and normal for getting the modid?

File: slsDetectorServers/eigerDetectorServer/slsDetectorServer_defs.h [hunk 2]
```diff
@@ -2,7 +2,11 @@
 #include "sls/sls_detector_defs.h"
 
 #define REQUIRED_FIRMWARE_VERSION (27)
-#define IDFILECOMMAND             "more /home/root/executables/detid.txt"
+#ifdef VIRTUAL
```
Comment (erikfrojdh) [inline slsDetectorServers/eigerDetectorServer/slsDetectorServer_defs.h:5]: Add a comment, why the filenames differ

**Review by erikfrojdh — COMMENTED** [review]
Looks good

File: slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c (hunk 1 above)
Comment (thattil) [reply slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c:305]: We can set module id only in Beb, which is not there in the virtual server. So, I just getModuleId as well but do not assign it to anything, as there is nothing to set for virtual servers.

Comment (thattil) [reply slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c:305]: Or should i set it to a variable which is picked up when streaming out udp packets?

File: slsDetectorServers/eigerDetectorServer/slsDetectorServer_defs.h (hunk 2 above)
Comment (thattil) [reply slsDetectorServers/eigerDetectorServer/slsDetectorServer_defs.h:7]: added the comment

**Pull request merged**